src/librustc_mir/transform/const_prop.rs

```diff
@@ -271,33 +273,34 @@ fn type_size_of<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
 }
 
 struct CanConstProp {
-    local: Local,
-    can_const_prop: bool,
+    can_const_prop: IndexVec<Local, bool>,
     // false at the beginning, once set, there are not allowed to be any more assignments
-    found_assignment: bool,
+    found_assignment: IndexVec<Local, bool>,
 }
 
 impl CanConstProp {
     /// returns true if `local` can be propagated
-    fn check<'tcx>(local: Local, mir: &Mir<'tcx>) -> bool {
+    fn check<'tcx>(mir: &Mir<'tcx>) -> IndexVec<Local, bool> {
         let mut cpv = CanConstProp {
-            local,
-            can_const_prop: true,
-            found_assignment: false,
+            can_const_prop: IndexVec::from_elem(true, &mir.local_decls),
+            found_assignment: IndexVec::from_elem(false, &mir.local_decls),
         };
+        for (local, val) in cpv.can_const_prop.iter_enumerated_mut() {
+            *val = mir.local_kind(local) == LocalKind::Temp;
+        }
```

Comment:
`rustc_trans` didn't have this limitation, so you should be able to check more.

Reply:
You told me to only accept temporaries ;)

Reply:
I forgot all the things I said but the only limitation we have is we can't *remove* `var = const` assignments, because debuginfo needs to see them - but you're not replacing anything.